packages/core/src/repository.ts

```diff
@@ -0,0 +1,46 @@
+import Document from './document';
+import DocID from '@ceramicnetwork/docid';
+
+export class Repository {
+  readonly #map: Map<string, Document> = new Map();
+
+  /**
+   * Stub for async loading of the document.
```

Comment:
Are you expecting that the document loading will actually happen within the Repository?  I had imagined the Repository just being focused on managing the caching and state persistence layer, without any of the actual business logic for loading a document.  I had hoped when a document isn't in the Repository and needs to be loaded, it would be loaded outside of the Repository and then provided to the Repository for caching/persistence once the load was complete.